zeppelin-interpreter/src/main/java/org/apache/zeppelin/resource/VFSResourcePool.java

```
@@ -0,0 +1,254 @@
+      }
+     
+      if (!isDirectory(resultDir)) {
+        throw new IOException(resultDir.getName().toString() + " is not a directory");
+      }
+      Gson gson = new Gson();
+
+      FileObject resultFile = resultDir.resolveFile("result.dat", NameScope.CHILD);
+      // false means not appending. creates file if not exists
+      OutputStream out = resultFile.getContent().getOutputStream(false);
+      ObjectOutputStream oos = new ObjectOutputStream(out);
+      oos.writeObject(object);
+      out.close();
+    } catch (IOException e) {
+      super.put(name, object);
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public Resource get(String name) {
+    return get(name, true);
+  }
+
+  @Override
+  public Resource get(String noteId, String paragraphId, String name) {
+    Resource r = get(noteId, paragraphId, name, true);
+    return new Resource(
+        new ResourceId(this.id(), noteId, paragraphId, name), r.get()); 
+  }
+
+  @Override
+  public Resource get(String name, boolean remote) {
+    try {
+      FileObject rootDir = getRootDir();
+      FileObject resultDir = rootDir.resolveFile(name, NameScope.CHILD);
+      if (!resultDir.exists() || !isDirectory(resultDir))
+        return null;
+      FileObject resultFile = resultDir.resolveFile("result.dat", NameScope.CHILD);
+
+      InputStream instream = resultFile.getContent().getInputStream();
+      ObjectInputStream ois = new ObjectInputStream(instream);
+
+      try {
+        Object o = ois.readObject();
+        String[] splitName = name.split("___");
+        ResourceId r = null;
+        if (splitName.length == 3)
+        {
+          r = new ResourceId(this.id(), splitName[0], splitName[1], splitName[2]);
+        }
+        else {
+          r = new ResourceId(this.id(), name);
+        }
+        return new Resource(r, o);
+      } catch (ClassNotFoundException e) {
+        throw new RuntimeException(e);
+      }
+
+    } catch (IOException e) {
+    }
```

logger.error("", e) if we are swallowing an exception?